src/core/utils/platform.ts

```diff
@@ -32,6 +64,39 @@ const normalizeTimestamp = (value?: string | Date) => {
   return Math.floor(timestamp / SECONDS_IN_MILLISECOND);
 };
 
+const normalizeDirection = (
+  direction?: string,
+): TransactionDirection | undefined => {
+  if (!direction) return undefined;
+  const normalized = direction.toLowerCase();
+  if (normalized === 'in' || normalized === 'out' || normalized === 'self') {
+    return normalized as TransactionDirection;
+  }
+  return undefined;
+};
+
+const normalizeTransactionType = (
+  type?: string,
+): TransactionType | undefined => {
+  if (!type) return undefined;
+  const normalized = type.toLowerCase();
+  return isValidTransactionType(normalized) ? normalized : undefined;
+};
+
+const normalizeStatus = (status?: string): TransactionApiResponse['status'] => {
+  if (!status) return 'pending';
+  const normalized = status.toLowerCase();
+  if (normalized === 'confirmed' || normalized === 'failed') {
+    return normalized;
+  }
+  return 'pending';
+};
+
+const toOptionalAddress = (value?: string) => {
+  if (!value || value.length === 0) return undefined;
+  return value as Address;
```

Comment:
Might consider using the Viem or ox util here instead of a type cast: https://oxlib.sh/api/Address/from#examples